tests/dat/actions/loggingTimeout.js

```diff
@@ -0,0 +1,100 @@
+     }
+     if (now - start >= testArgs.durationMillis) {
+        break;
+     }
+  }
+  return { iterations: i, logLines: logLines };
+}
+function task_cpu(testArgs) {
+   var func = (i, params) => Math.atan2(Math.sqrt(i), Math.sin(i + Math.cos(i)) * Math.cos(i + Math.sin(i)));
+   return doWork(func, testArgs);
+}
+function task_log(testArgs) {
+   return doWork(undefined, testArgs);
+}
+var tasks = {
+  task_cpu: task_cpu,
+  task_log: task_log
+}
+function getArg(value, d) {
+  if (value == undefined) {
+    return d;
+  }
+  return value;
+}
+function getRandomInt(min, max) {
+  return min + Math.floor(Math.random() * (max - min));
+}
+function main(args) {
+   var startMillis = new Date();
+   var testArgs = {
+      delayMillis: getArg(args.delayMillis, 0), // sleep time between iterations
+      durationMillis: getArg(args.durationMillis, 10000), // total duration of test (approx.)
+      addMillis: getArg(args.addMillis, 0), // added to durationMillis, but only in addMillisPercentage of the cases where this action is called
+      addMillisPercentage: getArg(args.addMillisPercentage, 0), // percentage of the cases where addMillis is added to durationMillis when this action is called
+      logDeltaMillis : getArg(args.logDeltaMillis, 100), // approx. time between two log lines
+      taskName : getArg(args.taskName, "task_log"), // name of task for the test (task_XXX)
+      taskParams : getArg(args.taskParams, undefined) // additional params for the task (if supported, ignored otherwise)
+   }
+   console.log("taskName: " + testArgs.taskName);
+   console.log("delayMillis: " + testArgs.delayMillis);
+   console.log("durationMillis: " + testArgs.durationMillis);
+   console.log("addMillis: " + testArgs.addMillis);
+   console.log("addMillisPercentage: " + testArgs.addMillisPercentage);
+   console.log("logDeltaMillis: " + testArgs.logDeltaMillis);
+   console.log("taskParams: " + testArgs.taskParams);
+   // check if addMillis should be added to durationMillis this time (random seed is based on system time)
+   if (getRandomInt(1, 100) <= testArgs.addMillisPercentage) {
+      testArgs.durationMillis += testArgs.addMillis;
+      console.log(">>> durationMillis updated to " + testArgs.durationMillis);
+   }
+   var taskFunc = tasks[testArgs.taskName];
+   var info = taskFunc(testArgs);
+   var totalMillis = new Date() - startMillis;
+   return { greetings : `[js] hello, I am back after ${totalMillis} ms`, iterations: info.iterations, logLines: info.logLines };
+}
+// following code is only active when being called with nodejs from command-line
+//console.log("process.env: " + JSON.stringify(process.env));
+if (! process.env.__OW_API_HOST) {
+  var params = '{}'
+  if (process.argv.length >= 3) {
+    params = process.argv[2];
+  }
+  info = main(JSON.parse(params));
+  console.log(JSON.stringify(info));
+}
```

Comment:
This function looks extremely complex. Can we tune it down to only do the minimally necessary work here?

Reply:
We have replaced the test action with a simple nodejs:6 version only using basic parameters (duration and delay)